machine-learning/sagemaker-edge-manager/create_components.py

```diff
@@ -0,0 +1,255 @@
+
+
+def upload_artifacts(file_name, object_name=None):
+    if object_name is None:
+        object_name = file_name
+    try:
+        response = s3_client.upload_file(file_name, bucket, object_name)
+    except Exception as e:
+        print(
+            "Failed to upload the artifacts to the bucket {} with key {}.\nException: {}".format(
+                bucket, object_name, e
+            )
+        )
+        exit(1)
+    print(
+        "Successfully uploaded the artifacts to the bucket {} with key {}.".format(
+            bucket, object_name
+        )
+    )
+
+
+def create_components():
+    for component_recipe in os.listdir(build_recipes_path):
+        recipe_file_path = os.path.join(build_recipes_path, component_recipe)
+        with open(
+            recipe_file_path,
+        ) as f:
+            try:
+                response = greengrass_client.create_component_version(inlineRecipe=f.read())
+            except Exception as e:
+                print(
+                    "Failed to create the component using the recipe at {}.\nException: {}".format(
+                        recipe_file_path, e
+                    )
+                )
+                exit(1)
+
+
+def copy_sample_models(path, model_names, c_name):
+    try:
+        response = s3_client.head_bucket(Bucket=public_models_bucket)
+        if response is not None and response["ResponseMetadata"]["HTTPStatusCode"] == 200:
+            for model_name in model_names:
+                s3 = boto3.resource("s3")
+                public_model_key = "{}/{}".format(c_name, model_name)
+                source = {
+                    "Bucket": public_models_bucket,
+                    "Key": public_model_key,
+                }
+                destination = s3.Bucket(bucket)
+                try:
+                    destination_model_key = "{}/{}".format(path, model_name)
+                    destination.copy(source, destination_model_key)
+                    print(
+                        "Successfully copied the model artifact {} from {} to {}".format(
+                            model_name, public_models_bucket, bucket
+                        )
+                    )
+                except Exception as e:
+                    print(
+                        "Failed to copy the model artifact {} from {} to {}.\nException: {}".format(
+                            model_name, public_models_bucket, bucket, e
+                        )
+                    )
+                    exit(1)
+        else:
+            print("Error retrieving the model bucket {}".format(public_models_bucket))
+            exit(1)
+    except Exception as e:
+        print(
+            "Failed to copy the model artifacts. Please check if the models bucket {} exists and try again. \nException: {}".format(
+                public_models_bucket, e
+            )
+        )
+        exit(1)
+
+
+def get_account_number():
+    try:
+        response = iam_client.get_user()
```

Comment:
I don't think that user is correct here. You want https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sts.html#STS.Client.get_caller_identity